logic/src/commonTest/kotlin/com/wire/kalium/logic/data/featureConfig/FeatureConfigRepositoryTest.kt

```diff
@@ -0,0 +1,137 @@
+package com.wire.kalium.logic.data.featureConfig
+
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.util.shouldFail
+import com.wire.kalium.logic.util.shouldSucceed
+import com.wire.kalium.network.api.ErrorResponse
+import com.wire.kalium.network.api.featureConfigs.FeatureConfigApi
+import com.wire.kalium.network.api.featureConfigs.FeatureConfigResponse
+import com.wire.kalium.network.exceptions.KaliumException
+import com.wire.kalium.network.utils.NetworkResponse
+import io.mockative.Mock
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class FeatureConfigRepositoryTest {
+
+    @Test
+    fun whenFileSharingFeatureConfigSuccess_thenTheSuccessIsReturned() = runTest {
+        // Given
+        val (arrangement, featureConfigRepository) = Arrangement().arrange()
+        arrangement.withSuccessfulResponse()
+
+        // when
+        val result = featureConfigRepository.getFileSharingFeatureConfig()
+
+        // then
+        result.shouldSucceed { arrangement.expectedSuccess.value }
+        verify(arrangement.featureConfigApi)
+            .suspendFunction(arrangement.featureConfigApi::fileSharingFeatureConfig)
+            .wasInvoked(once)
+    }
+
+    @Test
+    fun whenFileSharingFeatureConfigFailWithOperationDeniedError_thenTheErrorIsPropagated() = runTest {
+        // Given
+        val (arrangement, featureConfigRepository) = Arrangement().arrange()
+        arrangement.withOperationDeniedErrorResponse()
+
+        // when
+        val result = featureConfigRepository.getFileSharingFeatureConfig()
+
+        // then
+        result.shouldFail { arrangement.operationDeniedFailExpected.value }
+
+        verify(arrangement.featureConfigApi)
+            .suspendFunction(arrangement.featureConfigApi::fileSharingFeatureConfig)
+            .wasInvoked(exactly = once)
+    }
+
+    @Test
+    fun whenFileSharingFeatureConfigFailWithNoTeamError_thenTheErrorIsPropagated() = runTest {
+        // Given
+        val (arrangement, featureConfigRepository) = Arrangement().arrange()
+        arrangement.withNoTeamErrorResponse()
+
+        // when
+        val result = featureConfigRepository.getFileSharingFeatureConfig()
+
+        // then
+        result.shouldFail { arrangement.noTeamFailExpected.value }
+
+        verify(arrangement.featureConfigApi)
+            .suspendFunction(arrangement.featureConfigApi::fileSharingFeatureConfig)
+            .wasInvoked(exactly = once)
+    }
```

Comment:
The idea is that you setup all the conditions of the arrangement, and then eventually you call to `arrange()`
You could also provide the error externally instead of hardcoding it internally on the arrangement.
```suggestion
        // Given
        val expectedError = mockedNoTeamError
        val (arrangement, featureConfigRepository) = Arrangement()
                   .withErrorResponse(mockedNoTeamError)
                   .arrange()

        // when
        val result = featureConfigRepository.getFileSharingFeatureConfig()

        // then
        result.shouldFail { arrangement.noTeamFailExpected.value }

        verify(arrangement.featureConfigApi)
            .suspendFunction(arrangement.featureConfigApi::fileSharingFeatureConfig)
            .wasInvoked(exactly = once)
    }
```